app/controllers/api/v2/users_controller.rb

```diff
@@ -26,11 +28,35 @@ def accept_terms
       def generate_and_send_magic_link
         skip_authorization
 
-        GenerateAndSendMagicLinkService.call(user: current_user)
+        # If user is not found current_user returns "HTTP Token: Access denied.\n"
+        current_user = authenticate_or_request_with_http_token do |token|
+          AuthenticationService.new(logger: Rails.logger).execute(token)
```

Comment:
can we split this service into smaller chunks?

one to actual do the jwt related stuff
the other the authenticate bit fetching from db etc

then we can get rid of this  `current_user.is_a?(User)`

Reply:
`generate_and_send_magic_link`

should this method even exist whereby its `skip_before_action :authenticate`
yet do the exact same thing with `authenticate_or_request_with_http_token` which is in here `authenticate`

Reply:
This actually reads really confusing

we have a `generate_and_send_magic_link` that grabs a `current_user` via `AuthenticationService` by bypassing any of the `authenticate`/`authorization` related 

check if the `current_user` is an actual `user` then do the actual `generate_and_send_magic_link` otherwise do `send_email_address_unrecognised_email`

Reply:
I've cleaned it up a bit. Hopefully it reads better now. There is an AuthenticationService that would be idea for moving the JWT decoding logic to however, it's not fit purpose in this instance. I'm not sure if there is value creating a separate service?